### DISCUSSION TITLE
GridBase error in ESM

### DISCUSSION THREAD

**Discussion by gfo2007** [dfaruque/Serenity.Extra#100] [Q&A] [answered]
Hi @dfaruque ,
Thank you once again for the Serenity.Extra project. I get the error in the image below anytime I want Grid class to inherit GridBase

I have also added the code.

```
import { Decorators, EntityGrid } from '@serenity-is/corelib';
import { RefHoleLocationColumns, RefHoleLocationRow, RefHoleLocationService } from '../../ServerTypes/Setup';
import { RefHoleLocationDialog } from './RefHoleLocationDialog';
import { GridBase } from "@/_Ext/Bases/GridBase"

@Decorators.registerClass('geotrack_alt.Setup.RefHoleLocationGrid')
export class RefHoleLocationGrid extends GridBase<RefHoleLocationRow, any> {
    protected getColumnsKey() { return RefHoleLocationColumns.columnsKey; }
    protected getDialogType() { return RefHoleLocationDialog; }
    protected getRowDefinition() { return RefHoleLocationRow; }
    protected getService() { return RefHoleLocationService.baseUrl; }

    constructor(container: JQuery) {
        super(container);
    }
}
```

![GridBase error](https://user-images.githubusercontent.com/14317420/233852872-2306ad48-ff5e-45f8-b434-d82a82df62e1.png)

Regards

**Comment by dfaruque** [comment, marked as answer]
your getIdProperty is missing I think. try put the following line
`        protected getIdProperty() { return RefHoleLocationRow.idProperty; }`

**Reply by gfo2007** [reply]
@dfaruque thanks for your quick response. It worked.

However, for some reason the Page Display Name is not picking what I have customized in RefHoleLocationRow.cs. How do I make GridBase see my custom name? Code and image are below.

```
[ConnectionKey("Default"), Module("Setup"), TableName("REF_HOLE_LOCATION")]
[DisplayName("Hole Locations"), InstanceName("Hole Location")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
```

![Page Title problem](https://user-images.githubusercontent.com/14317420/233954394-a157dcc1-36b8-4f3c-91b1-9aa55a8ab1e5.png)

**Reply by dfaruque** [reply]
instead of ` protected getIdProperty() { return RefHoleLocationRow.idProperty; }` please override `getRowType()` like the following
`        protected getRowType() { return RefHoleLocationRow; }`

**Reply by gfo2007** [reply]
thank you.
